### PR TITLE
MVP GraphQL API and DB migrations

### PR DESCRIPTION
**Pull request by MChartier**
Using gqlgen to generate project structure and API stubs from GraphQL schema.
Removing existing OpenAPI implementation and spec submodule.

Adding database migrations to drop existing POC database tables and
replace them with new database tables that we will use to build the MVP.
This change is motivated primarily by the need to switch from an integer
to a varchar data type for the PRIMARY key on the Users table; altering
a data type of a column that is also a foreign key for other tables was
complicated enough that it seemed better to just take the opportunity
to redesign the tables entirely.

#25 #24 #26

### CODE REVIEW

File: graph/schema.graphqls [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+}
+
+type FoodRecord {
+    user: User!
+    calories: Int!
+    label: String!
+    timestamp: String!
+}
+
+enum Sex {
```
Comment (SlyCaptainFlint) [inline graph/schema.graphqls:31]: might want to add "other" here to cover our bases

File: graph/schema.resolvers.go [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+	"log"
+	"strconv"
+
+	"github.com/Matterhorn-Apps/MatterhornApiService/database"
+	"github.com/Matterhorn-Apps/MatterhornApiService/graph/generated"
+	"github.com/Matterhorn-Apps/MatterhornApiService/graph/model"
+)
+
+func (r *mutationResolver) CreateUser(ctx context.Context, input model.NewUser) (*model.User, error) {
+	// Query the database for matching exercise records
```
Comment (SlyCaptainFlint) [inline graph/schema.resolvers.go]: comment incorrect?

File: graph/schema.graphqls (hunk 1 above)
Comment (MChartier) [reply graph/schema.graphqls:31]: This field is already optional on the user type, but we can add something to cover cases where a user has explicitly declined to specify a binary option for sex.

File: graph/schema.resolvers.go [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+	}
+
+	// Query the database for matching Food records
+	query := fmt.Sprintf("INSERT INTO food_records (user_id, label, calories) VALUES ('%s', '%s', %d);",
+		input.UserID, *input.Label, *input.Calories)
+	_, readErr := r.DB.Exec(query)
+	if readErr != nil {
+		if errCode, ok := database.TryExtractMySQLErrorCode(readErr); ok {
+			switch *errCode {
+			case 1452:
```
Comment (SlyCaptainFlint) [inline graph/schema.resolvers.go:80]: what is this integral case?

File: graph/schema.resolvers.go [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+		Label:    *input.Label,
+	}, nil
+}
+
+func (r *mutationResolver) SetCalorieGoal(ctx context.Context, input model.CalorieGoalInput) (*model.CalorieGoal, error) {
+	if input.Calories < 0 {
+		return nil, errors.New("Invalid calorie goal provided")
+	}
+
+	// Query the database for matching exercise records
```
Comment (SlyCaptainFlint) [inline graph/schema.resolvers.go]: comment incorrect

**Review by SlyCaptainFlint — APPROVED** [review]

**Review by SlyCaptainFlint — CHANGES_REQUESTED** [review]
Need to fix database queries to match the new column names

File: graph/schema.resolvers.go (hunk 2 above)
Comment (MChartier) [reply graph/schema.resolvers.go]: Will fix.

File: graph/schema.resolvers.go (hunk 3 above)
Comment (MChartier) [reply graph/schema.resolvers.go:80]: This is detecting specific error codes from MySQL. Previously this was used to determine status codes for REST responses. Now it doesn't really serve any particular purpose, but I'm inclined to keep it around for now - we can perhaps add more specific and helpful log or error messages in the future for these expected cases.

File: graph/schema.resolvers.go (hunk 4 above)
Comment (MChartier) [reply graph/schema.resolvers.go]: Will fix.

**Review by SlyCaptainFlint — APPROVED** [review]

**Pull request merged**